presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -512,16 +514,16 @@ public synchronized void truncateUnpartitionedTable(ConnectorSession session, St
 
     public synchronized Optional<List<String>> getPartitionNames(String databaseName, String tableName)
     {
-        return doGetPartitionNames(databaseName, tableName, Optional.empty());
+        return doGetPartitionNames(databaseName, tableName, null);
```

Comment:
Usually we don't pass / accept nulls. Please either use Optional, or better just simply pass the `TuppleDomain.all()` predicate.